### DISCUSSION TITLE
Welcome to IOS-XE-Ops Discussions!

### DISCUSSION THREAD

**Discussion by cober2019** [cober2019/IOS-XE-Ops#2] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by NaNaR32** [comment]
Hi,

Sorry for my question, I am junior developer on Python.
For my trainee I will work with catalyst 9000 serie with IOS-XE


I not be able to run or install your application.
Have you got some informations/help to launch your application?

I try with this tutorial
https://pythonspot.com/flask-web-app-with-python/

But I have a problem 
Traceback (most recent call last):
  File "D:\Git\iosxe\IOS-XE-Ops\run.py", line 5, in <module>
    from decouple import config
ImportError: cannot import name 'config' from 'decouple'
C:\Users\NaNaR32\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\decouple\__init__.py

I not understand where from config 

Thanks a lot.

**Reply by cober2019** [reply]
I'm here to help. Did you run python3 install requirements.txt in your virtual environment ? The last requirement is:

python-decouple

You can install by executing pip3 or pip install python-decouple

With Linux you run the command globally but windows you need to go to your python scripts folder in command prompt.

**Reply by NaNaR32** [reply]
Thanks, yes I install 

pip3 install flask~=1.1.2
pip3 install ipaddress~=1.0.23
pip3 install requests
pip3 install flask_migrate
pip3 install flask_login
pip3 install flask_wtf
pip3 install WTForms~=2.3.3
pip3 install SQLAlchemy~=1.3.19
pip3 install xmltodict~=0.12.0
pip3 install ncclient~=0.6.9
pip3 install Jinja2~=2.11.2
pip3 install lxml~=4.6.2
pip3 install netmiko~=3.3.0
pip3 install python-decouple

After that I launch 
python run.py and python3 run.py

(my_env) PS D:\Git\iosxe\IOS-XE-Ops> python3 run.py
Traceback (most recent call last):
  File "D:\Git\iosxe\IOS-XE-Ops\run.py", line 5, in <module>
    from decouple import config
ImportError: cannot import name 'config' from 'decouple' (C:\Users\alixt\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\decouple\__init__.py)

(my_env) PS D:\Git\iosxe\IOS-XE-Ops> python run.py
Traceback (most recent call last):
  File "D:\Git\iosxe\IOS-XE-Ops\run.py", line 5, in <module>
    from decouple import config
ImportError: cannot import name 'config' from 'decouple' (D:\Git\iosxe\my_env\lib\site-packages\decouple\__init__.py)

I think it's simple but I can't find it out

**Reply by cober2019** [reply]
I was able to replicate the issue. You may have decouple and python decouple install at the same time. Please try pip uninstall decouple and run the program again

IOS-XE-Ops-main/run.py", line 5, in <module>
    from decouple import config

pip uninstall decouple
Uninstalling decouple-0.0.7:
  Would remove:
 
Proceed (y/n)? y
  Successfully uninstalled decouple-0.0.7

Then I ran run.py

 * Serving Flask app "app" (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: on
 *

**Reply by NaNaR32** [reply]
Thank is better :-) 

But now I have an error with 
D:\Git\iosxe\my_env\lib\site-packages\flask_sqlalchemy\__init__.py:851: UserWarning: Neither SQLALCHEMY_DATABASE_URI nor SQLALCHEMY_BINDS is set. Defaulting SQLALCHEMY_DATABASE_URI to "sqlite:///:memory:".
  warnings.warn(
D:\Git\iosxe\my_env\lib\site-packages\flask_sqlalchemy\__init__.py:872: FSADeprecationWarning: SQLALCHEMY_TRACK_MODIFICATIONS adds significant overhead and will be disabled by default in the 
future.  Set it to True or False to suppress this warning.
  warnings.warn(FSADeprecationWarning(
 * Serving Flask app "app" (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: on
D:\Git\iosxe\my_env\lib\site-packages\flask_sqlalchemy\__init__.py:851: UserWarning: Neither SQLALCHEMY_DATABASE_URI nor SQLALCHEMY_BINDS is set. Defaulting SQLALCHEMY_DATABASE_URI to "sqlite:///:memory:".
  warnings.warn(
D:\Git\iosxe\my_env\lib\site-packages\flask_sqlalchemy\__init__.py:872: FSADeprecationWarning: SQLALCHEMY_TRACK_MODIFICATIONS adds significant overhead and will be disabled by default in the 
future.  Set it to True or False to suppress this warning.
  warnings.warn(FSADeprecationWarning(

the terminal stay a this step and not launch app.

I think in run.py i need to change 
     app.run(host='0.0.0.0')
and install an database no?

**Reply by cober2019** [reply]
You don't have to install the database. Is this prevent you from running the program? Just go to  http://127.0.0.1:5000 in your broswer

**Reply by NaNaR32** [reply]
Thank a lot!
It's running

Device: is the IP or the name of catalyst switch?

**Reply by cober2019** [reply]
Yup

**Reply by cober2019** [reply]
How did the program work out for you? It's old and need updating. That'd actually my next project

**Reply by NaNaR32** [reply]
I have not access from home office, I will try tomorow.
